aws/resource_aws_mq_broker.go

```diff
@@ -391,18 +403,27 @@ func resourceAwsMqBrokerUpdate(d *schema.ResourceData, meta interface{}) error {
 		if err != nil {
 			return err
 		}
+		rebootMarker++
 	}
 
 	if d.HasChange("user") {
 		o, n := d.GetChange("user")
-		err := updateAwsMqBrokerUsers(conn, d.Id(),
+		var err error
```

Comment:
This approach is a bit brittle, but it was the only way to reliably detect user changes without refactoring the MQ Broker user functions which seems a bit out of the scope of this PR. I'd be happy to look into alternative suggestions though!

Reply:
We're not familiar with an issue with `d.HasChange("user")` like this:

> d.HasChange("user") always reports a change

Because we would expect the acceptance testing to previously have reported a difference after apply testing (`ExpectNonEmptyPlan` defaults to `false` in the testing framework terms). The only thing I can think of here would be the `groups` attribute inside `user` potentially showing as a difference not reported in the plan output but still listed as a difference in Terraform's graph when it makes it to the Update function. That would be quite a bug find! Are you sure this wasn't related the previous logic always calling `RebootBroker` when `apply_immediately = true` before the reboot variable was added?